crates/dids/src/method/jwk.rs

```diff
@@ -0,0 +1,120 @@
+use crate::did::Did;
+use crate::method::{DidMethod, DidMethodError};
+use crate::resolver::DidResolutionResult;
+use async_trait::async_trait;
+use crypto::key::{Key, KeyType};
+use crypto::key_manager::KeyManager;
+use did_jwk::DIDJWK as SpruceDidJwkMethod;
+use ssi_dids::did_resolve::{DIDResolver, ResolutionInputMetadata};
+use ssi_dids::{DIDMethod, Source};
+use std::sync::Arc;
+
+/// Concrete implementation for a did:jwk DID
+pub struct DidJwk {
+    uri: String,
+    key_manager: Arc<dyn KeyManager>,
+}
+
+impl Did for DidJwk {
+    fn uri(&self) -> &str {
+        &self.uri
+    }
+
+    fn key_manager(&self) -> &Arc<dyn KeyManager> {
+        &self.key_manager
+    }
+}
+
+/// Options that can be used to create a did:jwk DID
+pub struct DidJwkCreateOptions {
+    pub key_type: KeyType,
+}
+
+#[async_trait]
+impl DidMethod<DidJwk, DidJwkCreateOptions> for DidJwk {
+    const NAME: &'static str = "jwk";
+
+    fn create(
+        key_manager: Arc<dyn KeyManager>,
+        options: DidJwkCreateOptions,
+    ) -> Result<DidJwk, DidMethodError> {
+        let key_alias = key_manager.generate_private_key(options.key_type)?;
+        let public_key =
+            key_manager
+                .get_public_key(&key_alias)?
+                .ok_or(DidMethodError::DidCreationFailure(
+                    "PublicKey not found".to_string(),
+                ))?;
+
+        let uri = SpruceDidJwkMethod
+            .generate(&Source::Key(&public_key.jwk()))
+            .ok_or(DidMethodError::DidCreationFailure(
+                "Failed to generate did:jwk".to_string(),
+            ))?;
+
+        Ok(DidJwk { uri, key_manager })
+    }
+
+    async fn resolve_uri(did_uri: &str) -> DidResolutionResult {
+        let input_metadata = ResolutionInputMetadata::default();
+        let (did_resolution_metadata, did_document, did_document_metadata) =
+            SpruceDidJwkMethod.resolve(did_uri, &input_metadata).await;
+
+        DidResolutionResult {
+            did_resolution_metadata,
+            did_document,
+            did_document_metadata,
+            ..Default::default()
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use ssi_dids::did_resolve::ERROR_INVALID_DID;
+
+    fn create_did_jwk() -> DidJwk {
+        let key_manager = Arc::new(crypto::key_manager::LocalKeyManager::new_in_memory());
+        let options = DidJwkCreateOptions {
+            key_type: KeyType::Ed25519,
+        };
+
+        DidJwk::create(key_manager, options).unwrap()
+    }
+
+    #[test]
+    fn create_produces_correct_uri() {
+        let did = create_did_jwk();
+        assert!(did.uri.starts_with("did:jwk:"));
+    }
+
+    #[tokio::test]
```

Comment:
So why `tokio`?

Reply:
I presume it has something to do with us actually testing an external network integration

Reply:
DidJwk resolution should not go over the network. Chad tells me that

> The #[tokio::test] attribute is specific to the tokio runtime and is used to mark an asynchronous test function. It indicates that the following function is an asynchronous test that should be executed using the tokio runtime.

Reply:
@andresuribe87 has it right. 

Resolution is generic across all DID methods, and resolution CAN be an asynchronous operation for certain methods. So the `DidMethod` trait defines the resolve function as asynchronous.

While `did:jwk` does not resolve over the network, the function must be asynchronous to conform to the `DidMethod` trait. 

We have a few options to test asynchronous functions:
1. Write a normal, synchronous, test. Whenever you need to call an asynchronous function within the test, block the thread until the asynchronous function completes.
2. Use `tokio` or `async-std` (amongst others). 

Tokio is much more widely adopted, so I went with that. Happy to alter though if there's any strong opinions another way!